src/api/v1beta1/properties.go

```diff
@@ -206,13 +206,65 @@ func (dk *DynaKube) PullSecret() string {
 	return dk.Name + PullSecretSuffix
 }
 
-// ActiveGateImage returns the ActiveGate image to be used with the dk DynaKube instance.
-func (dk *DynaKube) ActiveGateImage() string {
-	if dk.CustomActiveGateImage() != "" {
-		return dk.CustomActiveGateImage()
+func (dk *DynaKube) NeedsReadOnlyOneAgents() bool {
+	inSupportedMode := dk.HostMonitoringMode() || dk.CloudNativeFullstackMode()
+	return inSupportedMode && !dk.FeatureDisableReadOnlyOneAgent()
+}
+
+func (dk *DynaKube) NeedsCSIDriver() bool {
+	isAppMonitoringWithCSI := dk.ApplicationMonitoringMode() &&
+		dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver != nil &&
+		*dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver
+
+	isReadOnlyHostMonitoring := dk.HostMonitoringMode() &&
+		!dk.FeatureDisableReadOnlyOneAgent()
+
+	return dk.CloudNativeFullstackMode() || isAppMonitoringWithCSI || isReadOnlyHostMonitoring
+}
+
+func (dk *DynaKube) NeedAppInjection() bool {
+	return dk.CloudNativeFullstackMode() || dk.ApplicationMonitoringMode()
+}
+
+func (dk *DynaKube) InitResources() *corev1.ResourceRequirements {
+	if dk.ApplicationMonitoringMode() {
+		return &dk.Spec.OneAgent.ApplicationMonitoring.InitResources
+	} else if dk.CloudNativeFullstackMode() {
+		return &dk.Spec.OneAgent.CloudNativeFullStack.InitResources
 	}
+	return nil
+}
 
-	return dk.DefaultActiveGateImage()
+func (dk *DynaKube) OneAgentResources() *corev1.ResourceRequirements {
+	switch {
+	case dk.ClassicFullStackMode():
+		return &dk.Spec.OneAgent.ClassicFullStack.OneAgentResources
+	case dk.HostMonitoringMode():
+		return &dk.Spec.OneAgent.HostMonitoring.OneAgentResources
+	case dk.CloudNativeFullstackMode():
+		return &dk.Spec.OneAgent.CloudNativeFullStack.OneAgentResources
+	}
+	return nil
+}
+
+func (dk *DynaKube) NamespaceSelector() *metav1.LabelSelector {
+	return &dk.Spec.NamespaceSelector
+}
+
+func (dk *DynaKube) NodeSelector() map[string]string {
+	switch {
+	case dk.ClassicFullStackMode():
+		return dk.Spec.OneAgent.ClassicFullStack.NodeSelector
+	case dk.HostMonitoringMode():
+		return dk.Spec.OneAgent.HostMonitoring.NodeSelector
+	case dk.CloudNativeFullstackMode():
+		return dk.Spec.OneAgent.CloudNativeFullStack.NodeSelector
+	}
+	return nil
+}
+
+func (dk *DynaKube) ActiveGateImage() string {
+	return dk.Status.ActiveGate.ImageURI()
 }
 
 func (dk *DynaKube) DefaultActiveGateImage() string {
```

Comment:
yes, otherwise how would the `version` package access them ? (we could move that logic there, but that would be not nice, also we have to think about the troubleshoot script)